docs/architecture-overview.md

```diff
@@ -0,0 +1,124 @@
+
+## Context
+
+- prev arch
+  - optimized for flexibility
+  - this made it difficult to learn
+
+
+## Decision
+
+- new architecture follows same general flow
+- general reduction in complexity
+- see `differences from previous architecture` for more in depth description of changes made to reduce complexity
+
+
+## Architecture Overview
```

Comment:
NOTE TO REVIEWERS: [the formatted version of this](https://github.com/Baron-Severin/fenix/blob/5799-document-architecture/docs/architecture-overview-outline.md) may be easier to review